utils/fasm/test/test_fasm.cpp

```diff
@@ -168,6 +169,54 @@ TEST_CASE("match_lut_init", "[fasm]") {
     CHECK(match_lut_init("16'b0000000011111111", "16'b0000111100001111"));
 }
 
+std::string get_pin_feature (size_t inode) {
```

Comment:
Should have a comment saying what this function does.
Also, I prefer declaring even local routines at the top of the file (then we can turn on missing prototype warnings, if they aren't on already).  If it's a local helper routine, best to declare it static. So I suggest:
- Declare this static if it's only intended for this file.
- Declare it at the top of the file, and comment what it does there.
- Comment should say what inode is (an rr_node index, or you could rename it to irr_node) and what the string format is and why you want it.